File: 167.TwoSumInputarrayIsSorted/sum.py
class Solution(object):
    def twoSum(self, numbers, target):
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """
        if not numbers:
            return []

        head, tail = 0, len(numbers) - 1
        while head < tail:
            sum = numbers[head] + numbers[tail]
            if sum == target:
                return [head+1, tail+1]
            if sum < target:
                head += 1
            else:
                tail -= 1

        return []

if __name__ == '__main__':
    print Solution().twoSum([2,7,11,15], 9)

File: 167.TwoSumInputarrayIsSorted/sum.cc
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        vector<int> result;

        if (numbers.size() == 0) {
            return result;
        }

        int head = 0, tail = int(numbers.size()) - 1;
        while (head < tail) {
            int sum = numbers[head] + numbers[tail];
            if (sum == target) {
                result.push_back(head+1);
                result.push_back(tail+1);
                return result;
            }
            if (sum < target) {
                head++;
            }
            else {
                tail--;
            }
        }

        return result;
    }
};
